port/linux/omrvmem.c

```diff
@@ -1507,7 +1577,7 @@ omrvmem_find_valid_page_size(struct OMRPortLibrary *portLibrary, uintptr_t mode,
  * to the newly allocated memory. Returns NULL on failure.
  */
 static void *
-getMemoryInRangeForDefaultPages(struct OMRPortLibrary *portLibrary, struct J9PortVmemIdentifier *identifier, OMRMemCategory *category, uintptr_t byteAmount, void *startAddress, void *endAddress, uintptr_t alignmentInBytes, uintptr_t vmemOptions, uintptr_t mode)
+getMemoryInRangeWithMmap(struct OMRPortLibrary *portLibrary, struct J9PortVmemIdentifier *identifier, OMRMemCategory *category, uintptr_t byteAmount, void *startAddress, void *endAddress, uintptr_t alignmentInBytes, uintptr_t vmemOptions, uintptr_t mode, uintptr_t pageSize)
```

Comment:
This method needs to scrutinized more. It's now used for large pages as well, and some parts of it may not make sense any more. @dmitripivkine is more familiar with this, but one example that I can see is call to adviseHugepage at the end, which probably does not make sense, if pageSize is large. Indeed, that call does not exist in (what used to be) getMemoryInRangeForLargePages

Reply:
Yeah the 2 different ones that large pages didn't use to have is the one you mentioned `adviseHugepage` and the other one is the part where it tries to find a smartAddress with `findAvailableMemoryBlockNoMalloc`. For the `adviseHugepage` we could just check if large pages are being used, but not too sure about the smartAddress part. Maybe we could use it for large pages as well?

Reply:
Right, quick search (smartAddress) is another one. Let's just wait for @dmitripivkine...

Reply:
As a part of `OMRPORT_VMEM_ALLOC_QUICK` optimization (used for object heap and code cache) `findAvailableMemoryBlockNoMalloc()` opens `/proc/self/map` and reads it line by line in attempt to find range best fitted to requested memory size/location to try at that (`smartAddress`) place, fail immediately if not allocated. So this is an idea. This logic should be used everywhere is possible regardless page size if applicable

Reply:
And in general I believe all functionality applicable to (or available for) `mmap()` type of allocation should be applied regardless page size.

Reply:
Seems problem is current code use default/large page size checks instead of mmap()/shmget() type of allocation (use to be equivalent). But it is not correct any more. So we need to review existing logic from this point

Reply:
That aspect looks to me ok. At the top level (logic in omrvmem_reserve_memory_ex) is still page size driven, and as it dives deeper in, for large pages, looks at underlying OS mechanism what specifically to call/use. Any other logic in this regard that is affected?

Reply:
Well, it is ok if we can do it without large mess. I can not find anything specific in `shmget()` path.